### ISSUE TITLE
CA1062 false positive in case of ParamArray (VB.net)

### ISSUE THREAD

**Issue by oahrens** [dotnet/roslyn-analyzers#2528]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3

#### Diagnostic ID

[CA1062](https://docs.microsoft.com/de-de/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods?view=vs-2019)

#### Repro steps

1. Copy the following example into a new VB project:

```
Public Class TestClass

  Public Sub Test(ParamArray ByVal values As String())

    Debug.Print(values.Length.ToString(System.Globalization.CultureInfo.InvariantCulture))

  End Sub

End Class
```

2. Install nuget package Microsoft.CodeAnalysis.FxCopAnalyzer

#### Expected behavior
CA1062 should not fire. [Microsoft](https://docs.microsoft.com/de-de/dotnet/visual-basic/programming-guide/language-features/procedures/parameter-arrays) states: 

> When you call a procedure that defines a parameter array, you can supply the argument in any one of the following ways: Nothing — that is, you can omit the ParamArray argument. In this case, an empty array is passed to the procedure. You can also pass the Nothing keyword, with the same effect.

#### Actual behavior
There is a compiler warning for `values` in `values.Length`.

**Comment by Gav-Brown** [comment]
It looks like v2.9.3 (VS 2019) enables the old `Managed Binary Analysis` (FXCop) rules even though `RunCodeAnalysis` is `false` in the .csproj file and all the rules are disabled.

**Comment by mavasani** [comment]
@oahrens This seems like a valid bug that we will fix.
@Gav-Brown Can you please provide a repro? 2.9.3 analyzer package contains new rules (CA1062 being one of them) which are enabled by default. If you have an explicit ruleset entry disabling CA1062, the rule should not fire.

**Comment by Gav-Brown** [comment]
@mavasani Thanks. I assumed it was the old FXCop rules as the .ruleset in Visual Studio didn't show any new rules. I suspect the ruleset editor has a bug...

**Comment by mavasani** [comment]
> I assumed it was the old FXCop rules as the .ruleset in Visual Studio didn't show any new rules. I suspect the ruleset editor has a bug...

Ah, that is likely. Thanks for the clarification.

**Comment by mavasani** [comment]
https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19317.5+45537f7f should have the fix

**Comment by oahrens** [comment]
It's ok now with version 2.9.4-beta1, thanks a lot!

**Comment by mavasani** [comment]
@oahrens Seems this was an incorrect bug, which has led to https://github.com/dotnet/roslyn-analyzers/issues/3845. Following code throws a null reference exception:

```vb
Imports System

Module Program
    Sub Main(args As String())
        M(Nothing)
    End Sub

    Sub M(ParamArray x As Integer())
        Console.WriteLine(x.Length)
    End Sub
End Module
```

I am going to revert this change.

**Comment by oahrens** [comment]
After a temporary solution - same again in Microsoft.CodeAnalysis.NetAnalyzers version 5.0.1

**Comment by mavasani** [comment]
@oahrens Can you please clarify your comment? As I mentioned in my last comment, your original report was incorrectas param arrays can be null/Nothing, so CA1062 is expected on the original code that you provided.

**Comment by oahrens** [comment]
> 
> 
> @oahrens Can you please clarify your comment? As I mentioned in my last comment, your original report was incorrect as param arrays can be null/Nothing, so CA1062 is expected on the original code that you provided.

There is nothing to clarify other than that **you are completely right**. Sorry, I havn't read your comment before leaving mine!
